astropy/io/fits/hdu/compressed.py

```diff
@@ -1595,6 +1595,13 @@ def header(self):
                 image_header['EXTNAME'] == self._default_name):
             del image_header['EXTNAME']
 
+        # Remove the PCOUNT GCOUNT cards if the uncompressed header is
+        # from a primary HDU
+        if 'SIMPLE' in image_header and \
+                ('PCOUNT' in image_header or 'GCOUNT' in image_header):
+            del image_header['PCOUNT']
+            del image_header['GCOUNT']
```

Comment:
A bit late to review but won't this crash if for some reason only one of those keywords are in the header? Are they guaranteed to always appear in pairs?

p.s. If there is a follow up PR to address this, then maybe replace the `\` continuation with parenthesis too. If not, it is okay... just a nitpick.

Reply:
Thanks @pllim ! In fact I've just realized that `PCOUNT` and `GCOUNT` are always both in `image_header` because they are explicitly set above https://github.com/astropy/astropy/blob/0446f168dc6e34996482394f00770b52756b8f9c/astropy/io/fits/hdu/compressed.py#L1560-L1573
Should I open a new PR to check only on the presence of `SIMPLE`?